c/datasetjson.c

```diff
@@ -1630,6 +1640,395 @@ void respondWithDataset(HttpResponse* response, char* absolutePath, int jsonMode
+      pid_t pid = getpid();
+      zowelog(NULL, LOG_COMP_HTTPSERVER, ZOWE_LOG_DEBUG,
+        "ZSS1692I %s PID is %d\n", __FUNCTION__, pid);
+
+  HttpRequest *request = response->request;
+
+  if (!isDatasetPathValid(absolutePath)) {
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Invalid dataset name");
+    return;
+  }
+
+  DatasetName dsn;
+  DatasetMemberName memberName;
+  extractDatasetAndMemberName(absolutePath, &dsn, &memberName);
+  
+  DynallocDatasetName daDsn;
+  DynallocMemberName daMember;
+  memcpy(daDsn.name, dsn.value, sizeof(daDsn.name));
+  memcpy(daMember.name, memberName.value, sizeof(daMember.name));
+
+  /* new code to detect ENQ header begins */
+  
+  /* new code ends   */
+
+/* new code to enqueue dataset starts */
+#include "isgenq.h" 
+#include <ctype.h>
+#define RESOURCE_NOT_AVAILABLE      0x0404
+/*#define MASK_FFFF                   256 * 256 - 1 */
+/*#define MASK_FFFF                   0xFFFF */
+    struct dsn_member_tag {
+        char  dsn[44];
+        char  membername[8];
+        } dsn_member; /* dataset name plus member, the resource name to be locked */
+
+    RName rname_parm; /* resource name; the name of the resource to acquire; the dataset name + optional member */
+     
+    static const QName MajorQNAME  = {"SPFEDIT "}; 
+        
+    ENQToken lockToken;
+    int lockRC = 0, lockRSN = 0;
+
+    memset(dsn_member.dsn,' ',44);          /* space-fill */
+    memset(dsn_member.membername,' ',8);    /* space-fill */
+    
+    memcpy(dsn_member.membername,daMember.name,8);   /* copy in member */
+    memcpy(dsn_member.dsn,daDsn.name,44); /* copy in dsn */
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+      "ZSS1797I %s dsn_member\n<", __FUNCTION__); 
+    int i;
+    for(i=0; i<52; i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%c",*(dsn_member.dsn+i));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+
+    memcpy(rname_parm.value, &dsn_member, sizeof(dsn_member) ); /* copy in dsn+membername */
+    rname_parm.length = 52;
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+      "ZSS1809I %s rname_parm\n", __FUNCTION__);
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"<"); 
+    for(i=0; i<52+sizeof(unsigned char); i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%x",*(rname_parm.value+i-sizeof(unsigned char)));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"<"); 
+    for(i=0; i<52; i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%c",*(rname_parm.value+i));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+    
+/* test lock */
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, 
+      "ZSS1826I %s Try exclusive Lock\n", __FUNCTION__);
+    
+/*    printf("TryExclusiveLock\n");  waits forever? CONTENTIONACT=FAIL */
+    lockRC = isgenqTryExclusiveLock(
```

Comment:
Does this function wait forever? I recall attempting a similar implementation and noting such behavior from isgenq functions. This could potentially cause the ZSS server to hang

Reply:
Great spot, but when I looked later, I found CONTENTIONACT=FAIL on that call, so it doesn't wait.  I forgot to amend the comment, which I've now done.